### DISCUSSION TITLE
Bug with jumping out of focus

### DISCUSSION THREAD

**Discussion by chris-yooo** [neuefische/questions#67] [Web] [answered]
i got a useEffect that are triggered every 7 seconds and fetched my messages, but this is also my problem the focus from input field gone away after fetchMessage.

![Screenshot (15)](https://user-images.githubusercontent.com/55069641/184834276-06d702b6-e60c-4164-a0de-873eab24bee0.png)

i tryed to build in the useRef with an focus on the input field but its buggy idk.

to test some the .env is visible in my repo: https://github.com/chris-yooo/capstone-project/tree/autoget

live view my app running: http://chrisyooo.ddns.net:3000

**Comment by pixelass** [comment]
Don't execute the `setTimeout` in the top level of your component body.
This is a "side-effect" and should be done in a `useEffect`.

**Reply by chris-yooo** [reply]
i can put this in my useEffect  ?

**Reply by chris-yooo** [reply]
i need this timer to fetch new messages

**Comment by PReichetanz** [comment]
Besides Gregor's hint, you don't pass the `dontJump` as a prop to your `CHRAT` component in App.js, line 74, so the reference does not arrive at your input field.

**Reply by chris-yooo** [reply]
ok i fixed that, but i need help to write the useEffect accordingly.

**Comment by actyvystom** [comment, marked as answer]
So, instead of using `setTimeout` you can implement a `useEffect` hook like this:
```javascript
useEffect(() => {
    async function fetchMessages() {
      try {
        const fetchedMessages = await getMessages();
        setMessages(fetchedMessages);
        // add your code to focus your input element
        
      } catch (err) {
        console.log(err);
      } finally {
        setShouldUpdate(false);

        jumpTo('anchor');
      }
    }
    const fetchInterval = setInterval(fetchMessages, 7000);
    return () => {
      clearInterval(fetchInterval);
    };
  }, []);
```

It creates the interval on the initialization of the component and clears the interval once the component is removed (to prevent staggering of multiple intervals, thus the return statement is important).
